internal/dag/dag.go

```diff
@@ -93,6 +93,17 @@ type RetryPolicy struct {
 	PerTryTimeout time.Duration
 }
 
+// UpstreamValidation defines how to validate the certificate on the upstream service
+type UpstreamValidation struct {
+	// Certificate holds a reference to the Secret containing the CA to be used to
+	// verify the upstream connection.
+	Certificate *Secret
```

Comment:
I think it would be good for the for the field name to be clear that it's a _CA_ certificate.

Reply:
Yeah I agree that more clear.